R/functions_io.R

```diff
@@ -294,36 +300,40 @@ ReadSparseMatrix = function(path, project="SeuratProject", row_name_column=2, co
   
   # create Seurat object with first assay
   # include cell and feature metadata
-  f = feature_types[1]
-  a = feature_type_to_assay_name[f]
-  sc = Seurat::CreateSeuratObject(counts=feature_data[[f]], 
+  sc = list()
+  f = feature_types[1] # feature type
+  a = feature_type_to_assay_name[f] # cell data
+  n = project # project name
+  m = metadata_table[, -1, drop=FALSE] # metadata for cells, also drop first column which contains cell names
+
+  sc[[n]] = Seurat::CreateSeuratObject(counts=feature_data[[f]], 
```

Comment:
Yes for Smartseq2 we can have multiple samples with cells in a table and then we return each sample as a separate dataset. For 10x, this is in theory possible but I have never seen such case. It could be the case when there is a big study and the data of multiple samples is returned in one big 10x matrix.

I will add an issue but so far it is not worth the effort.

Reply:
And why did you change it, compared to the way it was before (one object with all data)?